src/feature/dirauth/process_descs.c

```diff
@@ -121,6 +128,40 @@ add_fingerprint_to_dir(const char *fp, authdir_config_t *list,
   return;
 }
 
+/** Add the ed25519 key <b>edkey</b> to the smartlist of fingerprint_entry_t's
+ * <b>list</b>, or-ing the currently set status flags with
+ * <b>add_status</b>.
+ */
+/* static */ void
+add_ed25519_to_dir(const char *edkey, authdir_config_t *list,
```

Comment:
nickm says: (Also, passing around untyped arrays of bytes is questionable; it's probably best that anything that needs an ed25519 key should take the ed25519_public_key_t structure)